sources/packages/backend/apps/api/src/services/application-change-request/application-change-request.service.ts

```diff
@@ -0,0 +1,161 @@
+  INVALID_APPLICATION_EDIT_STATUS,
+} from "@sims/services/constants";
+import { CustomNamedError } from "@sims/utilities";
+
+/**
+ * Service responsible for application change request operations.
+ */
+@Injectable()
+export class ApplicationChangeRequestService {
+  constructor(
+    private readonly dataSource: DataSource,
+    @InjectRepository(Application)
+    private readonly applicationRepo: Repository<Application>,
+    private readonly noteSharedService: NoteSharedService,
+    private readonly workflowClientService: WorkflowClientService,
+    private readonly applicationService: ApplicationService,
+  ) {}
+
+  /**
+   * Assess the application change request status for the given application change request for the ministry user.
+   * @param applicationId application id for which to update the status.
+   * @param applicationEditStatus the application edit status to be updated.
+   * @param note note to be saved.
+   * @param auditUserId user that should be considered the one that is causing the changes.
+   */
+  async updateApplicationChangeRequestStatus(
+    applicationId: number,
+    applicationEditStatus:
+      | ApplicationEditStatus.ChangedWithApproval
+      | ApplicationEditStatus.ChangeDeclined,
+    note: string,
+    auditUserId: number,
+  ): Promise<void> {
+    // Get the application by id.
+    const changeRequestApplication =
+      await this.applicationService.getApplicationById(applicationId, {
+        allowEdited: true,
+      });
+    if (!changeRequestApplication) {
+      throw new CustomNamedError(
+        `Application ${applicationId} to assess change not found.`,
+        APPLICATION_NOT_FOUND,
+      );
+    }
+    if (
+      changeRequestApplication.applicationEditStatus !==
+      ApplicationEditStatus.ChangePendingApproval
+    ) {
+      throw new CustomNamedError(
+        `Application ${applicationId} to assess change not in valid status to be updated.`,
+        INVALID_APPLICATION_EDIT_STATUS,
+      );
+    }
+    const auditUser = { id: auditUserId } as User;
+    const currentDate = new Date();
+    await this.dataSource.transaction(async (transactionalEntityManager) => {
+      // Save the note.
+      await this.noteSharedService.createStudentNote(
+        changeRequestApplication.student.id,
+        NoteType.Application,
+        note,
+        auditUserId,
+        transactionalEntityManager,
+      );
+      const applicationRepo =
+        transactionalEntityManager.getRepository(Application);
+      if (applicationEditStatus === ApplicationEditStatus.ChangeDeclined) {
+        // Update the application edit status.
+        const updateResult = await applicationRepo.update(
+          {
+            id: applicationId,
+            applicationEditStatus: ApplicationEditStatus.ChangePendingApproval,
+          },
+          {
+            applicationEditStatus: ApplicationEditStatus.ChangeDeclined,
+            modifier: auditUser,
+            updatedAt: currentDate,
```

Comment:
Following audit properties are missing to be updated. 

`applicationEditStatusUpdatedBy`
`applicationEditStatusUpdatedOn`